qiskit_experiments/framework/base_analysis.py

```diff
@@ -32,17 +36,107 @@ class BaseAnalysis(ABC):
     For example, an analysis may perform some data processing of the
     measured data and a fit to a function to extract a parameter.
 
-    When designing Analysis subclasses default values for any kwarg
-    analysis options of the `run` method should be set by overriding
-    the `_default_options` class method. When calling `run` these
-    default values will be combined with all other option kwargs in the
-    run method and passed to the `_run_analysis` function.
+    Analysis subclasses must implement the abstract method `_run_analysis`.
+    This method should not have side-effects on the analysis class itself
+    since it could potentially be called asynchronously in multiple threads.
+    Any configurable option values should be specified in the `_default_options`
+    class method. These values can be overriden by a user by calling the
+    `set_options` method or for a single-run can be specified by passing kwarg
+    options to the :meth:`run` method.
     """
 
+    def __init__(self):
+        """Initialize the analysis object."""
+        # Analysis options
+        self._options = self._default_options()
+
+        # Store keys of non-default options
+        self._set_options = set()
+
+    def __new__(cls, *args, **kwargs):
```

Comment:
Will this be replaced with mix-in after #520 ?

Reply:
Yep